get_code on Cairo1 contract

**Describe the bug (observed vs expected behavior)**
run get_code endpoint for Cairo1 contract will produce internal server error:
```bash
[2023-05-04 17:29:19 +0000] [10] [ERROR] Exception on /feeder_gateway/get_code [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.9/site-packages/asgiref/sync.py", line 240, in __call__
    return call_result.result()
  File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.9/site-packages/asgiref/sync.py", line 306, in main_wrap
    result = await self.awaitable(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/starknet_devnet/blueprints/feeder_gateway.py", line 200, in get_code
    code_dict = await state.starknet_wrapper.get_code(contract_address, block_id)
  File "/usr/local/lib/python3.9/site-packages/starknet_devnet/starknet_wrapper.py", line 698, in get_code
    "bytecode": contract_class["program"]["data"],
KeyError: 'program'
```

**Not reproducible on alpha-goerli**

- [ ] This issue is only present on Devnet and cannot be reproduced on alpha-goerli (check the box if true).

**To Reproduce**
Steps to reproduce the behavior:

1. deploy cairo1 contract
2. run feeder_gateway/get_code?contractAddress=0x77f2....
   ...

**Devnet version**
shardlabs/starknet-devnet:0.5.1-seed0

- I am using Devnet version:
- [x] This happens with a dockerized Devnet (check the box if true).
- This does not appear on the following Devnet version:

**System specifications**

- OS:
- CPU:
- Memory:
- Python version:

--- a/starknet_devnet/__init__.py
+++ b/starknet_devnet/__init__.py
@@ -49,6 +49,9 @@ def _patch_poseidon_hash():
     starkware.cairo.common.poseidon_hash.poseidon_hash_many = getattr(
         poseidon_hash, "poseidon_hash_many"
     )
+    starkware.cairo.common.poseidon_hash.poseidon_hash_single = getattr(
+        poseidon_hash, "poseidon_hash_single"
+    )
     starkware.cairo.common.poseidon_hash.poseidon_perm = getattr(
         poseidon_hash, "poseidon_perm"
     )

--- a/starknet_devnet/starknet_wrapper.py
+++ b/starknet_devnet/starknet_wrapper.py
@@ -363,10 +363,10 @@ class StarknetWrapper:
             # extract class hash here if execution later fails
             class_hash = tx_handler.internal_tx.class_hash
 
-            # this is done now (before excute_tx) so that later we can assert it hasn't been deployed
+            # this is done now (before execute_tx) so that later we can assert it hasn't been deployed
             compiled_class_hash = await state.state.get_compiled_class_hash(class_hash)
 
-            # check if Declare v2
+            # check if Cairo 1 / Declare v2
             if isinstance(external_tx, Declare):
                 await assert_not_declared(class_hash, compiled_class_hash)
                 compiled_class_hash = tx_handler.internal_tx.compiled_class_hash
@@ -392,7 +392,7 @@ class StarknetWrapper:
                     ClassHashPair(class_hash, compiled_class_hash)
                 )
 
-            else:  # Cairo 0.x class
+            else:  # Cairo 0.x / Declare v1
                 tx_handler.execution_info = await state.execute_tx(
                     tx_handler.internal_tx
                 )
@@ -693,9 +693,14 @@ class StarknetWrapper:
         """Return code dict given the contract address"""
         try:
             contract_class = await self.get_class_by_address(contract_address, block_id)
+            bytecode = (
+                contract_class["program"]["data"]  # cairo 0
+                if "program" in contract_class
+                else contract_class["sierra_program"]  # cairo 1
+            )
             result_dict = {
                 "abi": contract_class["abi"],
-                "bytecode": contract_class["program"]["data"],
+                "bytecode": bytecode,
             }
         except StarkException as err:
             if err.code != StarknetErrorCode.UNINITIALIZED_CONTRACT:

--- a/test/test_declare_v2.py
+++ b/test/test_declare_v2.py
@@ -27,6 +27,7 @@ from .shared import (
 )
 from .test_state_update import get_state_update
 from .util import (
+    assert_contract_code_present,
     assert_hex_equal,
     assert_tx_status,
     assert_undeclared_class,
@@ -229,7 +230,7 @@ def test_v2_contract_interaction():
 
 @pytest.mark.declare
 @devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS)
-def test_v2_get_full_contract():
+def test_getting_contract_artifacts():
     """Test for declare, deploy and get full contract"""
 
     contract_class, _, compiled_class_hash = load_cairo1_contract()
@@ -257,7 +258,11 @@ def test_v2_get_full_contract():
         tx_hash=deploy_info["tx_hash"], expected_tx_status="ACCEPTED_ON_L2"
     )
 
+    # get_full_contract
     full_contract = get_full_contract_raw(contract_address=deploy_info["address"])
     assert full_contract.status_code == 200
     sierra = ContractClass.load(full_contract.json())
     assert load_sierra(CONTRACT_1_PATH) == sierra
+
+    # get_code
+    assert_contract_code_present(deploy_info["address"])

